tests/transforms/test_split_non_commuting.py

```diff
@@ -122,765 +119,580 @@ def complex_wires_processing_fn(results):
+            [qml.X(0)],
+            measurements=[qml.expval(qml.X(0)), qml.expval(qml.Y(1) + qml.Z(2))],
+            shots=100,
+        )
+        tapes, fn = split_non_commuting(initial_tape, grouping_strategy=grouping_strategy)
+        assert len(tapes) == 1
+        assert tapes[0].measurements == [
+            qml.expval(qml.X(0)),
+            qml.expval(qml.Y(1)),
+            qml.expval(qml.Z(2)),
         ]
-        for actual_tape, expected_tape in zip(tapes, expected_tapes):
-            qml.assert_equal(actual_tape, expected_tape)
+        assert tapes[0].shots.total_shots == 100
+        assert tapes[0].operations == [qml.X(0)]
+        assert qml.math.allclose(fn([[0.1, 0.2, 0.3]]), [0.1, 0.2 + 0.3])
 
-        result = ([0.1, 0.2], 0.2, 0.3, 0.4)
-        assert fn(result) == ((0.1, 0.2, 0.2), (0.3, 0.4))
+    @pytest.mark.parametrize("grouping_strategy", ["wires", "qwc", None])
+    def test_single_observable(self, grouping_strategy):
+        """Tests a tape containing measurements of a single observable."""
 
-    @pytest.mark.parametrize(
-        "non_pauli_obs",
-        [
-            [
-                qml.Projector([0], wires=[1]),
-                qml.Projector([1, 1, 0, 1], wires=[0, 1]),
-            ],
-            [
-                qml.Hadamard(wires=[1]),
-                qml.Hadamard(wires=[0]) @ qml.PauliX(wires=[1]),
+        initial_tape = qml.tape.QuantumScript(
+            [qml.X(0)], measurements=[qml.expval(qml.Z(0)), qml.expval(0.5 * qml.Z(0))], shots=100
+        )
+        tapes, fn = split_non_commuting(initial_tape, grouping_strategy=grouping_strategy)
+        assert len(tapes) == 1
+        assert tapes[0].measurements == [qml.expval(qml.Z(0))]
+        assert tapes[0].shots.total_shots == 100
+        assert tapes[0].operations == [qml.X(0)]
+        assert qml.math.allclose(fn([0.1]), [0.1, 0.05])
+
+    def test_mix_measurement_types_qwc(self):
+        """Tests multiple measurement types can be handled by qwc grouping"""
+
+        initial_tape = qml.tape.QuantumScript(
+            [qml.X(0)],
+            measurements=[
+                # [Z(0), X(0), Z(0), X(0), Y(1), Z(1), Z(0) @ Z(1), Z(1), Z(0)]
+                qml.expval(qml.Z(0)),
+                qml.expval(qml.Z(0) + qml.X(0)),
+                qml.var(qml.Z(0)),
+                qml.probs(op=qml.X(0)),
+                qml.counts(qml.Y(1)),
+                qml.sample(qml.Z(1)),
+                qml.probs(wires=[0, 1]),
+                qml.sample(wires=[1]),
+                qml.counts(wires=[0]),
             ],
-        ],
-    )
-    def test_tape_with_non_pauli_obs(self, non_pauli_obs):
-        """Tests that the tape is split correctly when containing non-Pauli observables"""
-
-        obs_list = single_term_obs_list + non_pauli_obs
+        )
 
-        measurements = [
-            qml.expval(c * o) for c, o in zip([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7], obs_list)
+        # ((0, 2, 5, 6, 7, 8), (1, 3, 4))
```

Comment:
Same as above -- is this relevant to the statements below?

Reply:
Also added a comment